client/src/components/User/UserBeaconSettings.vue

```diff
@@ -0,0 +1,285 @@
+            <div>
+              <b-button @click="optIn" variant="success">Enable</b-button>
+            </div>
+          </div>
+        </b-alert>
+
+        <div v-if="enabled">
+          <p>
+            You can share data by copying VCF or VCF.bgzip files to a history called
+            <span class="cursive gray-background">{{ beaconHistoryName }}</span>. <br>
+            <br>
+            The Beacon database is rebuilt periodically. Therefore, changes do not go into effect immediately.
+            If you disable beacon sharing or remove a dataset from the beacon history, the corresponding variants will
+            disappear from the beacon dataset during the next rebuild.
+          </p>
+        </div>
+
+
+        <!-- Detailed information about the beacon history -->
+        <div v-if="enabled" class="gray-box">
+          <!-- Case: History does not exist-->
+          <div v-if="beaconHistories.length < 1" class="flex-row history-entry">
+            <div class="no-shrink">
+              No beacon history found
+            </div>
+            <div class="fill"></div>
+            <div class="no-shrink">
+              <b-button @click="createBeaconHistory">Create Beacon History</b-button>
+            </div>
+          </div>
+
+          <!-- Case: History exists -->
+          <div v-for="beaconHistory in beaconHistories" :key="beaconHistory.id" class="flex-row history-entry"
+               :class="{'gray-border-bottom': beaconHistory.id !== beaconHistories[beaconHistories.length-1].id}">
+            <div class="no-shrink" v-if="beaconHistory.contents">
+              History with {{ beaconHistory.contents.length }} datasets
+            </div>
+            <div class="fill"></div>
+            <div class="no-shrink">
+              <b-button @click="switchHistory(beaconHistory.id)">Switch to History</b-button>
+            </div>
+          </div>
+        </div>
+
+        <div v-if="enabled">
+          <p>
+            Datasets must fulfill the following conditions in order to be processed
+          </p>
+          <ul>
+            <li>must be VCF or VCF.bgzip format</li>
+            <li>must have a human reference assigned to it (e.g. hg19)</li>
+            <li>must define at least one sample in a dedicated genotype column</li>
+            <li>must contain the info field <span class="cursive">AC</span>, with the total
+              number of alternate alleles in called genotypes
+            </li>
+          </ul>
+        </div>
+      </b-modal>
+    </div>
+  </b-row>
+</template>
+
+<script>
+import store from "store"
+import axios from "axios";
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+
+Vue.use(BootstrapVue);
+
+export default {
+  props: {
+    root: {
+      type: String,
+      required: true,
+    },
+    userId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      enabled: false,
+      beaconHistoryName: "Beacon Export 📡",
```

Comment:
Really cool!!
I first doubted that the emoji wouldn't break anything, but it seems to work just fine.